### PR TITLE
Expanding the intro a little

### PR DESCRIPTION
**Pull request by martinthomson**

### CODE REVIEW

File: draft-thomson-saag-sig-context.md [hunk 1]
```diff
@@ -42,25 +42,35 @@ have different semantics.  This produces a potential problem whereby signatures
...
 
 # Introduction
 
-There are a handful of cryptographic primitives that are used to build
-protocols that are used on the Internet: digital signatures, key derivation
-algorithms, message integrity codes, etc..  These protocols are generally
-developed in isolation from each other, with minimal effort to ensure that data
-structures used in one protocol do not have plausible interpretations in other
+Digital signatures are used in a range of different contexts.  These uses are
+often developed in isolation, which leads to the potential for data structures
```
Comment (dkg) [inline draft-thomson-saag-sig-context.md]: This reduces the scope to digital signatures, dropping key derivation algorithms and message integrity codes.  Do we want to narrow it in this way?

**Comment by martinthomson** [comment]
I was in two minds about this myself, but it turned out that going over this to broaden the focus didn't hurt that much.  Currently, I have dsig, kdf and mac.  If we want to expand further, that would be OK with me.  Note that we could recommend the use of labels with AEAD (the AAD), but it seems like the general guidance on key separation trumps that.

More advice on when to use key separation vs. labels might be good.

File: draft-thomson-saag-sig-context.md [hunk 2]
```diff
@@ -47,25 +47,35 @@ have different semantics.  This produces a potential problem whereby signatures
...
+
+Including a unique protocol- and usage- specific context label as input to all
+signing operations prevents a signature created in one context from being
+successfully validated in a different context.
+
+This document describes a uniform approach for the inclusion of context labels
+and a registry for unique labels.
+
+Existing protocols might already include a unique context label.  This document
+collects some of these existing labels into the context label registry.
```
Comment (kaduk) [inline draft-thomson-saag-sig-context.md:83]: Existing labels not collected by this document can also be added to the registry later, of course ... but that probably belongs in the IANA considerations and not in the introduction here.

**Comment by kaduk** [comment]
LGTM.
I agree that expanding the scope for HMAC and KDF is not very disruptive.

**Comment by martinthomson** [comment]
Addressed dkg's comment, will merge manually, since there are conflicts here.

**Pull request merged**